### PR TITLE
Adding mysql 5.7 and 8.0 keywords and functions for auto-completion

### PR DESCRIPTION
**Pull request by laoshaw**
## Description

Added keywords and json functions from mysql 5.7 and 8.0 for auto-completion

## Checklist

- [ ] I've added this contribution to the `changelog.md`.  -- not seen this file
- [ ] I've added my name to the `AUTHORS` file (or it's already there). -- change too small to be added

### CODE REVIEW

**Comment by tsroten** [comment]
@laoshaw Thanks for this pull request! We really appreciate this. I'm going to leave a few comments on this for some things that might need updating.

File: mycli/sqlcompleter.py [hunk 1]
```diff
@@ -37,11 +37,23 @@ class SQLCompleter(Completer):
                 'SLAVE', 'SMALLINT', 'SMALLINT', 'START', 'STOP', 'TABLE',
                 'THEN', 'TINYINT', 'TO', 'TRANSACTION', 'TRIGGER', 'TRUNCATE',
                 'UNION', 'UNIQUE', 'UNSIGNED', 'UPDATE', 'USE', 'USER',
-                'USING', 'VALUES', 'VARCHAR', 'VIEW', 'WHEN', 'WHERE', 'WITH']
+                'USING', 'VALUES', 'VARCHAR', 'VIEW', 'WHEN', 'WHERE', 'WITH',
+                'JSON', 'ISNULL', 'GREATEST', 'LEAST', 'BLOB', 'BIT', 'OPAQUE',
+                'DATETIME', 'BOOLEAN', 'ARRAY', 'OBJECT', 'STRING', 'DOUBLE', 'NULL',
+                'ENUM', 'TEXT']
```
Comment (tsroten) [inline mycli/sqlcompleter.py]: Can we move ISNULL, GREATEST, and LEAST down to the function list?

File: mycli/sqlcompleter.py [hunk 2]
```diff
@@ -37,11 +37,23 @@ class SQLCompleter(Completer):
...
                  'FROM_UNIXTIME', 'LAST', 'LCASE', 'LEN', 'MAX', 'MID',
-                 'MIN', 'NOW', 'ROUND', 'SUM', 'TOP', 'UCASE', 'UNIX_TIMESTAMP']
+                 'MIN', 'NOW', 'ROUND', 'SUM', 'TOP', 'UCASE', 'UNIX_TIMESTAMP',
+                 'JSON_APPEND', 'JSON_ARRAY', 'JSON_ARRAY_APPEND', 'JSON_ARRAY_INSERT',
+                 'JSON_CONTAINS', 'JSON_CONTAINS_PATH', 'JSON_DEPTH', 'JSON_EXTRACT',
+                 'JSON_INSERT', 'JSON_KEYS', 'JSON_LENGTH', 'JSON_MERGE',
+                 'JSON_MERGE_PATCH', 'JSON_MERGE_PRESERVE', 'JSON_OBJECT', 'JSON_PRETTY',
+                 'JSON_QUOTE', 'JSON_REMOVE', 'JSON_REPLACE', 'JSON_SEARCH', 'JSON_SET',
+                 'JSON_STORAGE_SIZE', 'JSON_TYPE', 'JSON_UNQUOTE', 'JSON_VALID',
+                 'JSON_ARRAYAVG', 'JSON_OBJECTAVG', 'JSON_OVERLAPS', 'JSON_SCHEMA_VALID',
```
Comment (tsroten) [inline mycli/sqlcompleter.py]: Did you mean `JSON_ARRAYAGG` and `JSON_OBJECTAGG`?

File: mycli/sqlcompleter.py [hunk 3]
```diff
@@ -37,11 +37,23 @@ class SQLCompleter(Completer):
...
+                 'JSON_SCHEMA_VALIDATION_REPORT', 'JSON_STORAGE_FREE', 'JSON_TABLE',
+                 'MEMBER_OF']
```
Comment (tsroten) [inline mycli/sqlcompleter.py]: `MEMBER_OF()` looks like it should be two words (`MEMBER OF()`).

**Comment by tsroten** [comment]
Also, do you think we could put the new suggestions in alphabetical order with the old ones? It definitely makes it easier for developers to find things in the list when we need to make changes, etc.

**Comment by amjith** [comment]
@laoshaw Are you still interested in addressing the comments that @tsroten left? 

If so, please let us know. I'll keep this PR open for a couple more days.

**Comment by laoshaw** [comment]
yes I think the new one is better.

**Comment by amjith** [comment]
> yes I think the new one is better.

I don't understand. Are you saying that you'll create a new PR or will you add more commits to this PR?

**Comment by pasenor** [comment]
Closing this, since it has not been updated since April, and now there is another take in #925 
@laoshaw feel free to open a new PR if you still want to work on the issue

**Pull request closed without merging**